Frameworks/CoreGraphics/CGColor.mm

```diff
@@ -14,73 +14,156 @@
 //
 //******************************************************************************
 
-#import <StubReturn.h>
-#import <Starboard.h>
-#import <CoreGraphics/CGColor.h>
-
-#import "UIColorInternal.h"
+#include <StubReturn.h>
+#include <Starboard.h>
+#include <CoreGraphics/CGColor.h>
+#include <CFCppBase.h>
+#include "CGColorInternal.h"
 
 const CFStringRef kCGColorBlack = CFSTR("BLACK");
 const CFStringRef kCGColorWhite = CFSTR("WHITE");
 const CFStringRef kCGColorClear = CFSTR("CLEAR");
 
-static IWLazyClassLookup __LazyUIColor("UIColor");
-
 static const wchar_t* TAG = L"CGColor";
 
+struct __CGColor : CoreFoundation::CppBase<__CGColor> {
+    // Override to add Equals method
+    static CFTypeID GetTypeID() {
+        static __CFRuntimeClass _cls{
+            0, // Version
+            typeid(__CGColor).name(), // Class Name
+            nullptr, // Init
+            nullptr, // Copy (ill-defined for CF)
+            &_CFFinalize, // Finalize
+            &__CFEquals, // Equals
+            nullptr, // Hash
+            nullptr, // Copy Description
+            nullptr, // Copy Debug Description
+        };
+        static CFTypeID _id = _CFRuntimeRegisterClass(&_cls);
+        return _id;
+    }
+
+    __CGColor(CGColorSpaceRef colorspace, CGFloat r, CGFloat g, CGFloat b, CGFloat alpha)
+        : _colorSpace(colorspace), _pattern(nullptr), _components{ r, g, b, alpha } {
+    }
+
+    __CGColor(CGColorSpaceRef colorSpace, CGPatternRef pattern) : _colorSpace(colorSpace), _pattern(pattern) {
+        // TODO #2581: CGColorCreateWithPattern should support components
+    }
+
+    // CreateCopyWithAlpha
+    __CGColor(const __CGColor& color, float alpha) : _colorSpace(color._colorSpace), _pattern(color._pattern) {
+        if (color.IsPatternColor()) {
+            _components.Clear();
+            _components.a = alpha;
+        } else {
+            _components.SetColorComponents(color._components.r, color._components.g, color._components.b, alpha);
+        }
+    }
+
+    bool operator==(const __CGColor& other) const {
+        return (CGColorSpaceGetModel(_colorSpace) == CGColorSpaceGetModel(other._colorSpace)) && (_pattern == other._pattern) &&
```

Comment:
that's not the scope of this PR. What we have is sufficient. 


---
In reply to: [113503142](https://github.com/Microsoft/WinObjC/pull/2538#discussion_r113503142) [](ancestors = 113503142)